inprocess/src/main/java/io/grpc/inprocess/InProcessTransport.java

```diff
@@ -414,6 +418,31 @@ private void streamClosed() {
       }
     }
 
+    private int getKnownLength(InputStream inputStream) throws IOException {
+      if (inputStream instanceof KnownLength || inputStream instanceof ByteArrayInputStream) {
```

Comment:
We need to support all InputStream types. These two conditions are the fast path. But we will still need the slow path that serializes into byte[]s and then creates a new InputStream for the receiving side. That is probably easiest done with `ByteStreams.toByteArray()` and `ByteArrayInputStream` (it isn't the most efficient, but in general if someone wants something to be fast we need KnownLength).